SWIG/date.i

```diff
@@ -419,10 +419,38 @@ using QuantLib::DateParser;
 
 #if defined(SWIGPYTHON)
 %{
+#if defined(Py_LIMITED_API)
+
+static PyObject* pydate_type;
+static PyObject *pydate_yearstr, *pydate_monthstr, *pydate_daystr;
+
+static inline bool PyDate_Check(PyObject* obj) {
+    return PyObject_TypeCheck(obj, (PyTypeObject*)pydate_type);
+}
+
+static inline PyObject* PyDate_FromDate(int year, int month, int day) {
+    return PyObject_CallFunction(pydate_type, "iii", year, month, day);
+}
+
+#define PyDateTime_GET_YEAR(o)  PyLong_AsLong(PyObject_GetAttr(o, pydate_yearstr))
+#define PyDateTime_GET_MONTH(o) PyLong_AsLong(PyObject_GetAttr(o, pydate_monthstr))
+#define PyDateTime_GET_DAY(o)   PyLong_AsLong(PyObject_GetAttr(o, pydate_daystr))
```

Comment:
Sorry, I noticed only after merging—`PyObject_GetAttr` returns a new reference, and we're discarding it after the conversion, so we should decref it after we're done, right?

Reply:
Good catch! Yes, you are absolutely right. Sorry about that. Here's the fix: https://github.com/lballabio/QuantLib-SWIG/pull/630